Documentation/op-guide/configuration.md

```diff
@@ -22,13 +22,13 @@ To start etcd automatically using custom settings at startup in Linux, using a [
 + This value is referenced as this node's own entries listed in the `--initial-cluster` flag (e.g., `default=http://localhost:2380`). This needs to match the key used in the flag if using [static bootstrapping][build-cluster]. When using discovery, each member must have a unique name. `Hostname` or `machine-id` can be a good choice.
 
 ### --data-dir
-+ Path to the data directory.
++ Path to the data directory. If this location does not exist, *etcd* will assume it has the required write permissions and attempt to create it.
 + default: "${name}.etcd"
 + env variable: ETCD_DATA_DIR
 
 ### --wal-dir
-+ Path to the dedicated wal directory. If this flag is set, etcd will write the WAL files to the walDir rather than the dataDir. This allows a dedicated disk to be used, and helps avoid io competition between logging and other IO operations.
-+ default: ""
++ Path to the dedicated wal directory. If this flag is set, etcd will write the WAL files to the walDir rather than the dataDir. This allows a dedicated disk to be used, and helps avoid io competition between logging and other IO operations. Since *etcd* writes folders adjacent to this location, it needs write permissions to the directory that holds the specified location. If this location does not exist, *etcd* will assume it has the required write permissions and attempt to create it.
++ default: "${ETCD_DATA_DIR}/wal"
 + env variable: ETCD_WAL_DIR
```

Comment:
@Saloshp I am sorry if I was not clear. For each flag `--data-dir` and `--wal-dir` 

add: `If this directory does not exist, *etcd* will attempt to create it`

Reply:
Can I also add information about the "${ETCD_WAL_DIR}.tmp" that is being created?
That was the issue all along, wanted to save time for others..
How shall I phrase it ?

Reply:
@Saloshp as I noted before the details of this process are not necessary for this document or the user. By telling the user that etcd will attempt to create the directory if it does not exist means that etcd requires perms to do so. I agreed this was not explicitly documented and appreciate you taking care of it.

In regards to the tmp dir if you have the proper perms this just works. The full details of the process are really only for nerds like us interested :). If folks run into trouble or have questions there search will lead them to this PR and the connected issue will point them to the source.

Thanks again for your patience and the PR.

Reply:
OK I will make the fix as requested

Reply:
Since you are working on rebasing feel free to --amend your commit message to include the flags you are adjusting ;)